src/deriver/enqueue.py

```
@@ -0,0 +1,286 @@
+
+    # Use the get_db dependency to ensure proper transaction handling
+    async with tracked_db("message_enqueue") as db_session:
+        try:
+            # Determine if batch or single processing
+            if not payload:  # Empty list check
+                return
+            workspace_name = payload[0]["workspace_name"]
+            session_name = payload[0]["session_name"]
+
+            if session_name is None or workspace_name is None:
+                raise ValidationException("Session and workspace are required")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)